### PR TITLE
yor: init at 0.1.177

### PR DESCRIPTION
**Pull request by ivankovnatsky**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ivankovnatsky** [comment]
Decided to disable tests for now, failing in arm darwin:

```
yor> please make sure you have .so file in ../../../tests/yor_plugins/example. if not, run the following command:
yor> go build -gcflags="all=-N -l" -buildmode=plugin -o ../../../tests/yor_plugins/example/extra_tags.so ../../../tests/yor_plugins/example/*.go
yor> please make sure you have .so file in ../../../tests/yor_plugins/tag_group_example. if not, run the following command:
yor> go build -gcflags="all=-N -l" -buildmode=plugin -o ../../../tests/yor_plugins/tag_group_example/extra_tag_groups.so ../../../tests/yor_plugins/tag_group_example/*.go
yor> --- FAIL: Test_loadExternalTags (0.00s)
yor>     --- FAIL: Test_loadExternalTags/load_tags_plugins (0.00s)
yor>         runner_test.go:39:
yor>                    Error Trace:    runner_test.go:39
yor>                    Error:          Not equal:
yor>                                    expected: 2
yor>                                    actual  : 0
yor>                    Test:           Test_loadExternalTags/load_tags_plugins
yor>     --- FAIL: Test_loadExternalTags/load_tagGroups_plugins (0.00s)
yor>         runner_test.go:71:
yor>                    Error Trace:    runner_test.go:71
yor>                    Error:          Not equal:
yor>                                    expected: 1
yor>                                    actual  : 0
yor>                    Test:           Test_loadExternalTags/load_tagGroups_plugins
yor> panic: runtime error: index out of range [0] with length 0 [recovered]
yor>    panic: runtime error: index out of range [0] with length 0
yor> goroutine 8 [running]:
yor> testing.tRunner.func1.2({0x1064237c0, 0x140000565a0})
yor>    /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1526 +0x1c8
yor> testing.tRunner.func1()
yor>    /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1529 +0x384
yor> panic({0x1064237c0, 0x140000565a0})
yor>    /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/runtime/panic.go:884 +0x204
yor> github.com/bridgecrewio/yor/src/common/runner.Test_loadExternalTags.func2(0x140006031e0)
yor>    /private/tmp/nix-build-yor-0.1.177.drv-0/source/src/common/runner/runner_test.go:72 +0x494
yor> testing.tRunner(0x140006031e0, 0x1066ebca0)
yor>    /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1576 +0x10c
yor> created by testing.(*T).Run
yor>    /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1629 +0x368
yor> FAIL       github.com/bridgecrewio/yor/src/common/runner   0.435s
yor> FAIL
error: builder for '/nix/store/8n5l76xy77b92dil578mfbwyzcmq8dqf-yor-0.1.177.drv' failed with exit code 1;
       last 10 log lines:
       > panic({0x1064237c0, 0x140000565a0})
       >       /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/runtime/panic.go:884 +0x204
       > github.com/bridgecrewio/yor/src/common/runner.Test_loadExternalTags.func2(0x140006031e0)
       >       /private/tmp/nix-build-yor-0.1.177.drv-0/source/src/common/runner/runner_test.go:72 +0x494
       > testing.tRunner(0x140006031e0, 0x1066ebca0)
       >        /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1576 +0x10c
       > created by testing.(*T).Run
       >         /nix/store/p8xaj4w8cjxc4js2pm56k2r3vcws92rq-go-1.20.3/share/go/src/testing/testing.go:1629 +0x368
       > FAIL github.com/bridgecrewio/yor/src/common/runner   0.435s
       > FAIL
```

and on hetzner amd:

```
yor> --- PASS: TestOutputCrasher (0.00s)
yor> === RUN   TestTagGroupCrasher
yor> --- PASS: TestTagGroupCrasher (0.00s)
yor> === RUN   TestListTagsGroupCrasher
yor> --- PASS: TestListTagsGroupCrasher (0.00s)
yor> PASS
yor> ok         github.com/bridgecrewio/yor/src/common/clioptions       0.013s
yor> === RUN   TestNewGitService
yor> === RUN   TestNewGitService/Get_correct_organization_and_repo_name
yor> 2023/05/23 12:43:40 Get "https://github.com/bridgecrewio/terragoat.git/info/refs?service=git-upload-pack": dial tcp: lookup github.com on [::1]:53: read udp [::1]:43916->[::1]:53: read: connection refused
yor> FAIL       github.com/bridgecrewio/yor/src/common/gitservice       0.015s
yor> FAIL
error: builder for '/nix/store/qrab1m6hi858l96mzimwvsrjqsjzh1yp-yor-0.1.177.drv' failed with exit code 1;
       last 10 log lines:
       > --- PASS: TestTagGroupCrasher (0.00s)
       > === RUN   TestListTagsGroupCrasher
       > --- PASS: TestListTagsGroupCrasher (0.00s)
       > PASS
       > ok    github.com/bridgecrewio/yor/src/common/clioptions       0.013s
       > === RUN   TestNewGitService
       > === RUN   TestNewGitService/Get_correct_organization_and_repo_name
       > 2023/05/23 12:43:40 Get "https://github.com/bridgecrewio/terragoat.git/info/refs?service=git-upload-pack": dial tcp: lookup github.com on [::1]:53: read udp [::1]:43916->[::1]:53: read: connection refused
       > FAIL     github.com/bridgecrewio/yor/src/common/gitservice       0.015s
       > FAIL
```

This latest looks like a firewall or networking issue, though I don't have access to firewall management, can't proceed further.

File: pkgs/applications/networking/cluster/yor/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "bridgecrewio";
+    repo = pname;
+    rev = "${version}";
+    hash = "sha256-tOYRd3LxSlAvXCW89LAm4GWWukDBZhsgYIWYlEVKokE=";
+  };
+
+  vendorHash =
+    "sha256-ZeTjGmlu8LndD2DKNncPzlpECdvkOjfwaVvV6S3sL9E=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/yor/default.nix:18]:
```suggestion
  vendorHash = "sha256-ZeTjGmlu8LndD2DKNncPzlpECdvkOjfwaVvV6S3sL9E=";
```

File: pkgs/applications/networking/cluster/yor/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}:
+
+buildGoModule rec {
+  pname = "yor";
+  version = "0.1.177";
+
+  src = fetchFromGitHub {
+    owner = "bridgecrewio";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/yor/default.nix:13]:
```suggestion
    rev = version;
```

File: pkgs/applications/networking/cluster/yor/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "sha256-ZeTjGmlu8LndD2DKNncPzlpECdvkOjfwaVvV6S3sL9E=";
+
+  doCheck = false;
+
+  # https://github.com/bridgecrewio/yor/blob/main/set-version.sh
+  preBuild = ''
+    echo "Updating version file with new tag: ${version}"
+    echo "package common" > src/common/version.go
+    echo "" >> src/common/version.go
+    echo "const Version = \"${version}\"" >> src/common/version.go
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/yor/default.nix:27]:
```suggestion
    echo "
    package common
    
    const Version = \"${version}\"" > src/common/version.go
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**